### ISSUE TITLE
Suggestion for AGM branching concept

### ISSUE THREAD

**Issue by bux** [KoffeinFlummi/AGM#1539]
@KoffeinFlummi @commy2 @esteldunedain @CorruptedHeart 

Here are some suggestions on how we could start to make better use of branches. My company is using a similar model to great success.
- Add "develop" / "dev" branch as an integration branch for features
- Features and Bugfixes (> 3 to 5 lines of code) happen in either a fork or their own feature branch they use "dev" as their parent.
- Feature branches or pull requests are merged into the "develop" / "dev" branch
- once "dev" is ready for release, fork a release branch. This is kind of a beta version. Meanwhile development can continue. If needed, bugfixes and release related stuff (docs, comments, versionnumbers) are merged into the release branch and dev branch
- if the release branch is finshed it is merged into master and tagged (already done through githubs releases) and merged back into dev
- master always contains the latest released version
- if hotfixes for the release are needed (e.g. fixes for FFV) they get a seperate hotfix branch and get merged into master (master is then again tagged) and dev branch

The most prominent change compared to today's workflow would be working in dev branch instead of master. The other suggestions mostly happen before and during releases and are in there to be able to continue developing while releasing or maybe hotfixing a release.

I'd love hearing your thoughts about it.
